07_boolean_2/main.go

```diff
@@ -15,56 +15,136 @@ E.g.:
 */
 
 func main() {
-	fmt.Println(false && (false || true))
-	fmt.Println((false && false) || true)
+	// fmt.Println(false && (false || true))
+	// fmt.Println((false && false) || true)
+	logicalTable(true, true)
+	logicalTable(true, false)
+	logicalTable(false, false)
+	logicalTable(false, true)
+	strIntCheck("", 0)
+	strIntCheck("Gopher", 3)
+	strIntCheck("Git", 2)
+	strIntCheck("Go", 4)
+}
 
-	// TODO: Write a function that prints a truth table for all the following
-	// Operations:
-	//  - AND
-	//  - OR
-	//  - NAND (NOT AND)
-	//  - NOR  (NOT OR)
-	//  - XOR  (EXCLUSIVE OR)
-	// E.g.: logicalTable(true, true) would print:
-	// AND : true
-	// OR  : true
-	// NAND: false
-	// ...
-	// Call this function for all 4 combinations of true and false.
+// TODO: Write a function that prints a truth table for all the following
+// Operations:
+//   - AND
+//   - OR
+//   - NAND (NOT AND)
+//   - NOR  (NOT OR)
+//   - XOR  (EXCLUSIVE OR)
+//
+// E.g.: logicalTable(true, true) would print:
+// AND : true
+// OR  : true
+// NAND: false
+// ...
+// Call this function for all 4 combinations of true and false.
+func logicalTable(bool1 bool, bool2 bool) {
+	// Varible used
+	// var AND bool = bool1 && bool2
+	// fmt.Println(AND)
+	// Hardcoded print results
+	// if bool1 == true && bool2 == true {
+	// 	fmpt.Println("true")
+	// }
+	// Direct print statements
+	// AND
+	fmt.Println("AND", bool1 && bool2)
+	// OR
+	fmt.Println("OR", bool1 || bool2)
+	// NAND
+	if bool1 == true && bool2 == true {
+		fmt.Println("NAND", false)
+	} else {
+		fmt.Println("NAND", true)
+	}
+	//  NOR
+	if bool1 == false && bool2 == false {
+		fmt.Println("NOR", true)
+	} else {
+		fmt.Println("NOR", false)
+	}
+	// XOR
+	if bool1 == true && bool2 == false || bool2 == true && bool1 {
+		fmt.Println("XOR", true)
+	} else if bool1 == bool2 || bool2 == bool1 {
+		fmt.Println("XOR", false)
+	}
+}
 
-	// TODO: Write a function that takes a string and an integer and performs
-	// the following checks:
-	// - If the string is empty, return "empty"
-	// - If the string is longer than the integer
-	//   - If the integer is even, return "longer and even"
-	//   - If the integer is odd, return "longer and odd"
-	// - If the string is shorter than the integer, return "shorter"
+// TODO: Write a function that takes a string and an integer and performs
+// the following checks:
+// - If the string is empty, return "empty"
+// - If the string is longer than the integer
+//   - If the integer is even, return "longer and even"
+//   - If the integer is odd, return "longer and odd"
+//
+// - If the string is shorter than the integer, return "shorter"
+func strIntCheck(str string, num int) {
+	// var stringLength = len(str)
+	// var integerLength = len(num)
+	if str == "" {
+		fmt.Println("empty")
+		// TODO find way to get int length
+		// } else if len(str) > len(num) {
+		if len(str)%2 == 0 {
```

Comment:
@nobe4 Unsure currently how to calculate length of `int`.
At least without anther import.
Maybe some loop?

Reply:
does https://github.com/riraum/forward/pull/10/files#r1793296769 clarify the exercise?

Reply:
YES. I do appreciate the challenge, hahaha.